### PR TITLE
Feature/iia 998 pattern nav

### PR DESCRIPTION
**Pull request by dholubek**
image

### CODE REVIEW

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java [hunk 1]
```diff
@@ -32,23 +56,72 @@ public class ConceptPatternNavController {
...
+//                        }
+//
+//                        @Override
+//                        protected void updateItem(EntityFacade item, boolean empty) {
+//                            super.updateItem(item, empty);
+//
+//                            if (item == null || empty) {
+//                                setGraphic(null);
+//                            } else {
+//                                //will need the EntityFacade as the underlying value..???
```
Comment (swaroopsalvi) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java:108]: Do you need to keep this commented code?

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java [hunk 2]
```diff
@@ -61,25 +134,133 @@ private void showPatterns() {
...
-        //FIXME this is mocked up
-        VBox resultsVBox = new VBox();
-        IntStream.range(0, 3).forEachOrdered(n -> {
+    public void loadPatternsIntoPane(Map<EntityFacade, List<SemanticRecord>> patterns) {
+        if (patterns.isEmpty()) {
+            return;
+        }
+        patternNavViewModel.setPropertyValue(PATTERN_COLLECTION, patterns);
+
+        //VBox resultsVBox = new VBox(); //FIXME turn into a ListView
```
Comment (swaroopsalvi) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java:182]: Can we remove this comment?

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java [hunk 3]
```diff
@@ -61,25 +134,133 @@ private void showPatterns() {
...
-            if (n > 0) {
-                controller.setPatternName("Test Performed Pattern");
-            }
-            resultsVBox.getChildren().addAll(new HBox(leftPadding, pattern));
+
+
+//            resultsVBox.setSpacing(4); // space between pattern entries
+//            patternHBox.setAlignment(Pos.CENTER);
+//            Region leftPadding = new Region();
+//            leftPadding.setPrefWidth(12); // pad each entry with an empty region
```
Comment (swaroopsalvi) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java:197]: Remove commented code.

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java [hunk 4]
```diff
@@ -61,25 +134,133 @@ private void showPatterns() {
...
+//            leftPadding.setPrefHeight(1);
+
+            controller.setPatternName(pattern.description());
+
+//            controller.getInstancesVBox().getChildren().clear();
+//            controller.getInstancesVBox().getChildren().addAll(createInstances(listOfSemantics));
+
+
```
Comment (swaroopsalvi) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java:205]: Remove commented code.

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java [hunk 5]
```diff
@@ -61,25 +134,133 @@ private void showPatterns() {
...
+                                label.getStyleClass().add("search-entry-description-label");
+                                setGraphic(label);
+                            }
+                        }
+                    });
+
+                    Platform.runLater(() -> {
+                        patternInstances.setItems(patternChildren);
+                        //if (patternChildren.size() == 0) {
+                        //    patternNavEntryController.disableInstancesListView();
```
Comment (swaroopsalvi) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java:349]: Remove commented code.

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/viewmodel/PatternNavViewModel.java [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+//                    return NaturalOrder.compareStrings(viewProperties.calculator().getDescriptionTextOrNid(nid1),
+//                            viewProperties.calculator().getDescriptionTextOrNid(nid2));
+//                } else {
+//                    return NaturalOrder.compareStrings(o1.toString(), o2.toString());
+//                }
+//            });
+//            Platform.runLater(() -> this.rootTreeItem.getChildren().setAll(patternItems));
+//            for (TreeItem<Object> patternItem : patternItems) {
+//                ArrayList<TreeItem<Object>> patternChildren = new ArrayList<>();
+//                int patternNid = (Integer) patternItem.getValue();
```
Comment (swaroopsalvi) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/viewmodel/PatternNavViewModel.java:66]: Remove commented code.

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/PatternNavEntryController.java [hunk 7]
```diff
@@ -45,4 +54,17 @@ private void initialize() {
...
+    public ListView getPatternInstancesListView() {
+        return this.patternInstancesListView;
+    }
+
+    public void disableInstancesListView() {
+        patternInstancesListView.setVisible(false);
+        patternInstancesListView.setDisable(true);
+    }
+
+//    public VBox getInstancesVBox() {
```
Comment (swaroopsalvi) [inline kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/PatternNavEntryController.java:67]: Remove commented code.

**Review by swaroopsalvi — COMMENTED** [review]
@dholubek , Saw lot of commented code, I guess you have kept it for future reference. Instead can I recommend to keep it in a branch to avoid any commented code to go in to main branch ?

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java (hunk 3 above)
Comment (dholubek) [reply kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java:156]: yep, cleaning up now

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java (hunk 5 above)
Comment (dholubek) [reply kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/ConceptPatternNavController.java]: removed

File: kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/PatternNavEntryController.java (hunk 7 above)
Comment (dholubek) [reply kview/src/main/java/dev/ikm/komet/kview/mvvm/view/navigation/PatternNavEntryController.java]: removed

**Pull request merged**